### PR TITLE
feat(admin-ui): implement UMA option in Scopes form 

### PR DESCRIPTION
**Pull request by mjatin-dev**
https://github.com/GluuFederation/flex/issues/346

### CODE REVIEW

File: admin-ui/app/locales/en/translation.json [hunk 1]
```diff
@@ -169,6 +169,9 @@
     "saml1_uri": "Saml1 URI",
     "saml2_uri": "Saml2 URI",
     "scope_type": "Scope type",
+    "umaAuthorizationPolicies": "Authorization Policies",
+    "umaResourcesScript": "Associated Clients",
```
Comment (duttarnab) [inline admin-ui/app/locales/en/translation.json:173]: Why `umaResourcesScript` is `Associated Clients`? 

`Associated Clients`  should be obtained scopes obj passed from the list page.

image

File: admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js [hunk 2]
```diff
@@ -31,38 +31,67 @@ import GluuTooltip from 'Routes/Apps/Gluu/GluuTooltip'
...
+  let umaAuthorizationPolicies = []
+
   const theme = useContext(ThemeContext)
   const selectedTheme = theme.state.theme
   const history = useHistory()
   const spontaneousClientScopes = scope.attributes.spontaneousClientScopes || []
-   const dispatch = useDispatch()
-   const client = useSelector((state) => state.oidcReducer.items)
+  const dispatch = useDispatch()
+  const client = useSelector((state) => state.oidcReducer.items)
```
Comment (duttarnab) [inline admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js:48]: This is not required. Please populated associated clients for UMA and spontaneous scopes from scope obj.

File: admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js [hunk 3]
```diff
@@ -31,38 +31,67 @@ import GluuTooltip from 'Routes/Apps/Gluu/GluuTooltip'
...
     .filter((item) => item.scriptType == 'DYNAMIC_SCOPE')
     .filter((item) => item.enabled)
     .map((item) => ({ dn: item.dn, name: item.name }))
+
+  umaAuthorizationPolicies = scripts
+    .filter((item) => item.scriptType == 'UMA_RPT_POLICY')
+    .filter((item) => item.enabled)
+    .map((item) => ({ dn: item.dn, name: item.name }))
+
+  umaResourcesScript = umaResources.map((item) => ({
```
Comment (duttarnab) [inline admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js:64]: I think this is not needed now.

File: admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js [hunk 4]
```diff
@@ -31,38 +31,67 @@ import GluuTooltip from 'Routes/Apps/Gluu/GluuTooltip'
...
+    dn: item.dn,
+    name: item.id,
+  }))
+
+  useEffect(() => {
```
Comment (duttarnab) [inline admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js:69]: not needed.

File: admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js [hunk 5]
```diff
@@ -31,38 +31,67 @@ import GluuTooltip from 'Routes/Apps/Gluu/GluuTooltip'
...
+  )
+  const [showSpontaneousPanel, handleShowSpontaneousPanel] = useState(
+    scope.scopeType === 'spontaneous',
+  )
+  const [showUmaPanel, handleShowUmaPanel] = useState(scope.scopeType === 'UMA')
 
   useEffect(() => {
     if (showSpontaneousPanel) {
-      dispatch(searchClients({ "action_data": makeOptions(1, scope.attributes.spontaneousClientId) }))
+      dispatch(
```
Comment (duttarnab) [inline admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js:90]: This is not required. Please populated associated clients for UMA and spontaneous scopes from scope obj.

**Review by duttarnab — CHANGES_REQUESTED** [review]

File: admin-ui/app/locales/en/translation.json (hunk 1 above)
Comment (mjatin-dev) [reply admin-ui/app/locales/en/translation.json]: removed

File: admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js (hunk 2 above)
Comment (mjatin-dev) [reply admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js]: done

File: admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js (hunk 3 above)
Comment (mjatin-dev) [reply admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js]: done, deleted

File: admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js (hunk 4 above)
Comment (mjatin-dev) [reply admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js]: removed

File: admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js (hunk 5 above)
Comment (mjatin-dev) [reply admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js]: removed

File: admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js [hunk 6]
```diff
@@ -120,10 +133,10 @@ function ScopeForm({ scope, scripts, attributes, handleSubmit}) {
     document.getElementsByClassName('UserActionSubmitButton')[0].click()
   }
 
-  const goToClientViewPage = (client_id) => {
+  const goToClientViewPage = (client_id, client_data = {}) => {
```
Comment (duttarnab) [inline admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js:136]: please use camel case (not snake case).

File: admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js [hunk 7]
```diff
@@ -120,10 +133,10 @@ function ScopeForm({ scope, scripts, attributes, handleSubmit}) {
...
     dispatch(viewOnly(true))
-    dispatch(setCurrentItem(client[0]))
-    return history.push(`/auth-server/client/edit:` + client_id.substring(0, 4))
+    dispatch(setCurrentItem(client_data))
```
Comment (duttarnab) [inline admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js:138]: please use camel case (not snake case).

**Review by duttarnab — CHANGES_REQUESTED** [review]

**Review by duttarnab — APPROVED** [review]

**Pull request merged**